### ISSUE TITLE
The Abacus terminated unexpectedly during the SOC self-consistent calculation.

### ISSUE THREAD

**Issue by jingan-181** [deepmodeling/abacus-develop#5483]
### Describe the bug

When performing the SOC self-consistent calculation in Abacus using a parallel scheme with 1 process and 40 threads, Abacus encountered an Intel error message during the first electronic step. In debug mode, the running_scf.log file gets stuck at the **Charge_Mixing::mix_rho** function, and the memory is sufficient at this point, so memory issues can be ruled out. 

Furthermore, when SOC is disabled, Abacus completes all the self-consistent tasks without any issues.

I have encountered this bug with Abacus versions 3.8.0 to 3.8.2 on different machines.

[03_scf_pbe.zip](https://github.com/user-attachments/files/17744530/03_scf_pbe.zip)

### Environment

- OS: centos 7.0 
- Compiler: intel oneapi

### Task list for Issue attackers (only for developers)

- [ ] Verify the issue is not a duplicate.
- [ ] Describe the bug.
- [ ] Steps to reproduce.
- [ ] Expected behavior.
- [ ] Error message.
- [ ] Environment details.
- [ ] Additional context.
- [ ] Assign a priority level (low, medium, high, urgent).
- [ ] Assign the issue to a team member.
- [ ] Label the issue with relevant tags.
- [ ] Identify possible related issues.
- [ ] Create a unit test or automated test to reproduce the bug (if applicable).
- [ ] Fix the bug.
- [ ] Test the fix.
- [ ] Update documentation (if necessary).
- [ ] Close the issue and inform the reporter (if applicable).

**Comment by mohanchen** [comment]
Could you check old versions of ABACUS, such as 3.4, 3.5, 3.6, 3.7?

**Comment by WHUweiqingzhou** [comment]
@jingan-181 
I failed to reproduce this unexpected stop. My calculation enters SCF successfully.
```
                              ABACUS v3.8.3

               Atomic-orbital Based Ab-initio Computation at UStc                    

                     Website: http://abacus.ustc.edu.cn/                             
               Documentation: https://abacus.deepmodeling.com/                       
                  Repository: https://github.com/abacusmodeling/abacus-develop       
                              https://github.com/deepmodeling/abacus-develop         
                      Commit: 264915205 (Wed Nov 20 08:53:18 2024 +0800)

 Wed Nov 20 13:50:42 2024
 MAKE THE DIR         : OUT.scf/
 RUNNING WITH DEVICE  : CPU / Intel(R) Xeon(R) Platinum 8269CY CPU @ 2.50GHz

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
 Warning: the number of valence electrons in pseudopotential > 4 for Sn: [Kr] 4d10 5s2 5p2
 Pseudopotentials with additional electrons can yield (more) accurate outcomes, but may be less efficient.
 If you're confident that your chosen pseudopotential is appropriate, you can safely ignore this warning.
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

 UNIFORM GRID DIM        : 1440 * 1440 * 50
 UNIFORM GRID DIM(BIG)   : 288 * 288 * 25
 DONE(4.01289    SEC) : SETUP UNITCELL
 DONE(5.36694    SEC) : INIT K-POINTS
 ---------------------------------------------------------
 Self-consistent calculations for electrons
 ---------------------------------------------------------
 SPIN    KPOINTS         PROCESSORS  THREADS     NBASE       
 4       10              32          32          6080        
 ---------------------------------------------------------
 Use Systematically Improvable Atomic bases
 ---------------------------------------------------------
 ELEMENT ORBITALS        NBASE       NATOM       XC          
 Sn      2s2p2d1f-10au   25          80          
 S       2s2p1d-10au     13          80          
 ---------------------------------------------------------
 Initial plane wave basis and FFT box
 ---------------------------------------------------------
 DONE(7.81044    SEC) : INIT PLANEWAVE
 -------------------------------------------
 SELF-CONSISTENT : 
 -------------------------------------------
 START CHARGE      : atomic
 DONE(285.421    SEC) : INIT SCF
 ITER     TMAGX      TMAGY      TMAGZ       AMAG        ETOT/eV          EDIFF/eV         DRHO     TIME/s
 GE1      1.90e-15  -5.94e-16   1.28e-01   2.34e+01  -1.95530319e+05   0.00000000e+00   2.4152e-01 325.74
```

**Comment by WHUweiqingzhou** [comment]
Fixed by #5545
